### PR TITLE
BaSP-TG: Employee-put-deleted-filter

### PR DESCRIPTION
**Pull request by AlvaroGuardia**
Finished the new problematic, create the sentence of put, deleted and filter of the Employees.

### CODE REVIEW

**Review by AlfonsoDalix — COMMENTED** [review]
You should delete commented lines

File: src/resources/employees.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
+// Call to Server and File System
+
+const express = require('express');
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:3]: Cambiemos los const express = require('express'); 
por import express from 'express';

File: src/resources/employees.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+const fs = require('fs');
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:5]: Lo mismo que el comment de arriba

File: src/resources/employees.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+const router = express.Router();
+
+const employees = require('../data/employees.json');
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:9]: Lo mismo que arriba

File: src/resources/employees.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
+// Call to Server and File System
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:1]: Delete this line

File: src/resources/employees.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+const express = require('express');
+
+const fs = require('fs');
+
+const router = express.Router();
+
+const employees = require('../data/employees.json');
+
+// Get all of the list/JSON
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:11]: Delete this line

File: src/resources/employees.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+router.get('/getAll', (req, res) => {
+  res.status(200).json({
+    employees,
+  });
+});
+
+// Filter by Id
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:19]: Delete this line

File: src/resources/employees.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+  const employeesId = req.params.id;
+  const users = employees.find((employee) => employee.id === employeesId);
+  if (users) {
+    res.send(users);
+  } else {
+    res.send('user not found');
+  }
+});
+
+// Filter by Status
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:31]: Delete this line

File: src/resources/employees.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  const employeesStatus = req.params.active;
+  const users = employees.filter((employee) => employee.active.toString() === employeesStatus);
+  if (users) {
+    res.send(users);
+  } else {
+    res.send('user status not found');
+  }
+});
+
+// Filter by Full Name
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:43]: Delete this line

File: src/resources/employees.js [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+  const employeesFname = req.query.full_name;
+  const Fname = employees.filter((employee) => employee.full_name === employeesFname);
+  if (Fname.length > 0) {
+    res.send(Fname);
+  } else {
+    res.send('Full Name not found');
+  }
+});
+
+// Filter by Gender
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:55]: Delete this line

File: src/resources/employees.js [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+  const employeesGender = req.params.gender;
+  const Gndr = employees.filter((employee) => employee.gender === employeesGender);
+  if (Gndr.length > 0) {
+    res.send(Gndr);
+  } else {
+    res.send('Gender not found - Insert Female or Male by parameters');
+  }
+});
+
+// Put Sentence
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:67]: Delete this line

File: src/resources/employees.js [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+      } else {
+        res.send('Employee edit');
+      }
+    });
+  } else {
+    res.send('No Employee with the id selected');
+  }
+});
+
+// Delete Sentence
```
Comment (AlfonsoDalix) [inline src/resources/employees.js:103]: Delete this line

File: src/resources/employees.js (hunk 5 above)
Comment (AlfonsoDalix) [reply src/resources/employees.js:11]: const employees = require('../data/employees.json'); change for import .. from...

**Review by Alfondalix — APPROVED** [review]
Good job

**Review by AlfonsoDalix — APPROVED** [review]

**Pull request merged**